### PR TITLE
Fix Strapi requests from browser

### LINKED ISSUE #428: Incorrect operation of "router-link" in the MENU
When [running in docker](https://github.com/edmcouncil/html-pages?tab=readme-ov-file#run-with-docker), there is an issue with navigating between MENU items (upper right corner).

For example, when you are on the [FIBO Home](http://localhost:8080/fibo/) page, after selecting the [FIBO Community](http://localhost:8080/fibo/page/community) item from the menu, you are not taken to the http://localhost:8080/fibo/page/community page.

However, when you enter http://localhost:8080/fibo/page/community in the browser, the correct page [FIBO Community](http://localhost:8080/fibo/page/community) opens.

---

After enabling _Developer Tools_ in the browser, this can be seen when trying to select [FIBO Community](http://localhost:8080/fibo/page/community) from the menu, an attempt to access URL:
```
http://fibo-strapi:1337/api/pages?populate[0]=sections&populate[1]=sections.items&populate[2]=sections.items.image&filters[slug][$eq]=community
```
results in failure.

The correct address should be:
```
http://localhost:8080/fibo/strapi/api/pages?populate[0]=sections&populate[1]=sections.items&populate[2]=sections.items.image&filters[slug][$eq]=community
```

### PR DESCRIPTION
**Pull request by przemekgradzki**
Closes #428

### CODE REVIEW

**Review by mereolog — DISMISSED** [review]

**Review by rosolowski — CHANGES_REQUESTED** [review]
The `baseURL` function doesn't require `async` keyword because it doesn't return a Promise. The code can be simplified by removing `async`, and then `await` keywords from baseURL function calls in strapi.js.

File: general/api/strapi.js [hunk 1]
```diff
@@ -2,6 +2,10 @@ import axios from 'axios';
 import qs from 'qs';
 import fs from 'fs';
 
+async function baseURL(runtimeConfig) {
```
Comment (rosolowski) [inline general/api/strapi.js]:
```suggestion
function baseURL(runtimeConfig) {
```

File: general/api/strapi.js [hunk 2]
```diff
@@ -13,15 +17,14 @@ export async function getStrapiSingleType(singleTypeName, populateParams, runtim
   );
 
   let response = await axios.get(`/api/${singleTypeName}?${query}`, {
-    baseURL: runtimeConfig.public.strapiBaseUrl,
+    baseURL: await baseURL(runtimeConfig),
```
Comment (rosolowski) [inline general/api/strapi.js]:
```suggestion
    baseURL: baseURL(runtimeConfig),
```

File: general/api/strapi.js [hunk 3]
```diff
@@ -42,7 +45,7 @@ export async function getStrapiElementFromCollection(
   });
 
   let response = await axios.get(`/api/${collectionName}?${query}`, {
-    baseURL: runtimeConfig.public.strapiBaseUrl,
+    baseURL: await baseURL(runtimeConfig),
```
Comment (rosolowski) [inline general/api/strapi.js]:
```suggestion
    baseURL: baseURL(runtimeConfig),
```

File: general/api/strapi.js [hunk 4]
```diff
@@ -68,7 +71,7 @@ export async function getStrapiCollection(
   });
 
   const response = await axios.get(`/api/${collectionName}?${query}`, {
-    baseURL: runtimeConfig.public.strapiBaseUrl,
+    baseURL: await baseURL(runtimeConfig),
```
Comment (rosolowski) [inline general/api/strapi.js]:
```suggestion
    baseURL: baseURL(runtimeConfig),
```

File: general/api/strapi.js [hunk 5]
```diff
@@ -158,9 +161,7 @@ export async function getAppConfigurationData(runtimeConfig) {
 
     if (imgPath && runtimeConfig.public.staticGenerationMode) {
       const imageName = imgPath.split('/').pop();
-      const imageUrl = (typeof window !== 'undefined'
-        ? `${window.location.origin}/${runtimeConfig.public.strapiBasePath}`
-        : runtimeConfig.public.strapiBaseUrl) + imgPath;
+      const imageUrl = await baseURL(runtimeConfig) + imgPath;
```
Comment (rosolowski) [inline general/api/strapi.js]:
```suggestion
      const imageUrl = baseURL(runtimeConfig) + imgPath;
```

File: general/api/strapi.js [hunk 6]
```diff
@@ -230,9 +231,7 @@ async function tryToDownloadImages(
         const imageResponseUrl = item?.image?.data?.attributes?.url;
         if (imageResponseUrl != undefined) {
           const imageName = imageResponseUrl.split('/').pop();
-          const imageUrl = (typeof window !== 'undefined'
-            ? `${window.location.origin}/${runtimeConfig.public.strapiBasePath}`
-            : runtimeConfig.public.strapiBaseUrl) + imageResponseUrl;
+          const imageUrl = await baseURL(runtimeConfig) + imageResponseUrl;
```
Comment (rosolowski) [inline general/api/strapi.js]:
```suggestion
          const imageUrl = baseURL(runtimeConfig) + imageResponseUrl;
```

**Review by rosolowski — CHANGES_REQUESTED** [review]
Simplified code sugestion

**Review by rosolowski — APPROVED** [review]

**Review by mereolog — APPROVED** [review]

**Pull request merged**